File: NewsAlarm/app/src/main/java/com/alarm/newsalarm/database/AlarmDataDao.java
package com.alarm.newsalarm.database;

import androidx.room.Dao;
import androidx.room.Delete;
import androidx.room.Insert;
import androidx.room.Query;
import androidx.room.Update;

import java.util.List;

@Dao
public interface AlarmDataDao {

    @Insert
    void insert(AlarmData... data);

    @Update
    void update(AlarmData data);

    @Delete
    void delete(AlarmData data);

    @Query("SELECT * FROM AlarmDataTable")
    List<AlarmData> getAll();

    @Query("DELETE FROM AlarmDataTable")
    void clear();
}

File: NewsAlarmKt/app/src/main/java/com/alarm/newsalarmkt/database/AlarmDataDao.kt
package com.alarm.newsalarmkt.database

import androidx.room.Dao
import androidx.room.Delete
import androidx.room.Insert
import androidx.room.Query
import androidx.room.Update

@Dao
interface AlarmDataDao {
    @Insert
    fun insert(vararg data: AlarmData)

    @Update
    fun update(data: AlarmData)

    @Delete
    fun delete(data: AlarmData)

    @get:Query("SELECT * FROM AlarmDataTable")
    val all: MutableList<AlarmData>

    @Query("DELETE FROM AlarmDataTable")
    fun clear()
}
